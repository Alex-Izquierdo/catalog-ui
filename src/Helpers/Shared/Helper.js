import React from 'react';

export function bindMethods (context, methods) {
  methods.forEach(method => {
    context[method] = context[method].bind(context);
  });
};


