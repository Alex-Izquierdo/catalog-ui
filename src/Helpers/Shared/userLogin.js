import { AdminsApi } from 'service_portal_api';
import { DefaultApi } from 'topological_inventory';

// TODO - replace with actual login info when available
let admin_hash = { 'x-rh-auth-identity': btoa(JSON.stringify({ identity: { is_org_admin: true }})) };
const adminApi = new AdminsApi();
Object.assign(adminApi.apiClient.defaultHeaders, admin_hash);

export function getUserApi() {
  return adminApi;
}
//UserSecurity.username = 'YOUR USERNAME'
//UserSecurity.password = 'YOUR PASSWORD'

let userTopologicalApi = new DefaultApi();

export function getTopologicalUserApi() {
  return userTopologicalApi;
}
