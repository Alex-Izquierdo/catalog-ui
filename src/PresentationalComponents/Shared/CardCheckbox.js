import React from 'react';
import propTypes from 'prop-types';
import { Checkbox } from '@patternfly/react-core';

const CardCheckbox = ({ handleCheck, isChecked, id }) => (
  <div style={ { float: 'right' } }>
    <Checkbox
      checked={ isChecked }
      onChange={ handleCheck }
      onClick={ event => event.stopPropagation() }
      aria-label="card checkbox"
      id={ id }
    />
  </div>
);

CardCheckbox.propTypes = {
  handleCheck: propTypes.func,
  isChecked: propTypes.bool,
  id: propTypes.string
};

export default CardCheckbox;
