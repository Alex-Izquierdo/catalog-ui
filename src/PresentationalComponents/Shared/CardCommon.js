import React from 'react';
import PropTypes from 'prop-types';

const PropLine = ({ value }) => <div>{ value }</div>;

PropLine.propTypes = {
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.arrayOf(PropTypes.oneOfType([
      PropTypes.arrayOf(PropTypes.node),
      PropTypes.node
    ]))
  ]).isRequired
};

const ItemDetails = ({ toDisplay = [], ...item }) => (
  <div className="line-clamp">
    { toDisplay.map(prop => <PropLine key={ `card-prop-${prop}` } value={ item[prop] } />) }
  </div>
);

ItemDetails.propTypes = {
  toDisplay: PropTypes.arrayOf(PropTypes.string)
};

ItemDetails.defaultProps = {
  toDisplay: []
};

export default ItemDetails;
