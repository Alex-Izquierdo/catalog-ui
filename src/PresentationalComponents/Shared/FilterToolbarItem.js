import React, { Component } from 'react';
import propTypes from 'prop-types';
import { Section } from '@red-hat-insights/insights-frontend-components';
import { ToolbarGroup, ToolbarItem, TolbarSection, Title, Button, TextInput } from '@patternfly/react-core';
import { css } from '@patternfly/react-styles';

class FilterToolbarItem extends Component {
    state = {
        searchValue: ''
    };

    render() {
        const { searchValue } = this.state;
        return (
            <ToolbarGroup className="searchToolbar">
                <ToolbarItem>
                    <div className="pf-c-input-group">
                        <TextInput placeholder={ this.props.placeholder } value={ searchValue } type="text" onChange={ this.props.onFilterChange }
                            aria-label="Find product button"></TextInput>
                        <Button variant="tertiary" id="searchProductButton" onClick={ this.onSearchSubmit }>
                            <i className="fas fa-search" aria-hidden="true"></i>
                        </Button>
                    </div>
                </ToolbarItem>
            </ToolbarGroup>);
    };
};

FilterToolbarItem.propTypes = {
    history: propTypes.object,
    onFilterChange: propTypes.func,
    placeholder: propTypes.string
};

export default FilterToolbarItem;

