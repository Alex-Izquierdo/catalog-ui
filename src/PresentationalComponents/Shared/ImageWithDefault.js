import React from 'react';
import propTypes from 'prop-types';
import DefaultPortfolioImg from '../../assets/images/default-portfolio.jpg';

const ImageWithDefault = ({ src, ...props }) => <img src={ src } { ...props } />;

ImageWithDefault.propTypes = {
  src: propTypes.string
};

ImageWithDefault.defaultProps = {
  src: DefaultPortfolioImg
};

export default ImageWithDefault;
