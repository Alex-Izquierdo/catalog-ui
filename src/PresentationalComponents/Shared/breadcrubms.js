import React from 'react';
import { get } from 'lodash';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { NavLink, withRouter } from 'react-router-dom';
import { Breadcrumb, BreadcrumbItem } from '@patternfly/react-core';

const patternsReducers = {
  portfolios: 'portfolioReducer.selectedPortfolio.name',
  platforms: 'platformReducer.selectedPlatform.name',
  orders: 'ordersReducer'
};

const fragmentMapper = {
  portfolios: {
    title: 'Portfolios'
  },
  detail: {
    attribute: 'name'
  },
  'add-products': {
    title: 'Add products'
  },
  'remove-products': {
    title: 'Remove products'
  },
  platforms: {
    title: 'Platforms'
  },
  orders: {
    title: 'Orders'
  }
};

const findRoutes = (url) => {
  const fragments = (url.split('/')).filter(item => item !== '');
  const cleanFragments = fragments.map((key, index) => {
    return {
      reducer: patternsReducers[key],
      urlFragment: key,
      index,
      isId: !!key.match(/^[0-9]+$/)
    };
  });

  if (cleanFragments.length === 0) {
    return [];
  }

  const rootReducer = cleanFragments.find(({ reducer }) => reducer !== undefined).reducer;
  let finalFragments = [];
  let actualIndex = 0;
  cleanFragments.forEach(({ reducer, isId, urlFragment }) => {
    if (isId) {
      finalFragments[actualIndex - 1] = {
        ...finalFragments[actualIndex - 1],
        path: `${finalFragments[actualIndex - 1].path}/${urlFragment}`
      };
      /**
       * remove id from URL fragments
       */
      finalFragments = [
        ...finalFragments.slice(0, actualIndex),
        ...finalFragments.slice(actualIndex + 1)
      ];
    } else {
      finalFragments[actualIndex] = {
        reducer: reducer || rootReducer,
        path: actualIndex === 0 ? `/${urlFragment}` : `${finalFragments[actualIndex - 1].path}/${urlFragment}`,
        urlFragment
      };
      /**
       * Increase the real index of finalFragments
       */
      actualIndex++;
    }
  });
  return finalFragments.map(fragment => ({
    ...fragment,
    meta: fragmentMapper[Object.keys(fragmentMapper).find(key => fragment.urlFragment.includes(key))]
  }));
};

const CatalogBreadrubms = ({ match: { url }, location: { pathname }, reducers }) => {
  const routes = findRoutes(url);
  const items = routes.map((route, index) => (
    <BreadcrumbItem key={ route.path } isActive={ route.path === pathname || index === routes.length - 1 }>
      <NavLink exact to={ route.path }>
        { route.meta.title || get(reducers, route.reducer) }
      </NavLink>
    </BreadcrumbItem>
  ));
  return (
    <Breadcrumb>
      { items }
    </Breadcrumb>
  );
};

const mapStateToProps = state => ({
  reducers: { ...state }
});

CatalogBreadrubms.propTypes = {
  match: PropTypes.shape({
    url: PropTypes.string.isRequired
  }).isRequired,
  location: PropTypes.shape({
    pathname: PropTypes.string.isRequired
  }),
  reducers: PropTypes.object
};

export default withRouter(connect(mapStateToProps)(CatalogBreadrubms));
