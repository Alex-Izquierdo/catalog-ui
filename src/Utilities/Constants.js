export const DEFAULT_PAGE_SIZE = 50;

const topoApiHost = process.env.TOPO_API_HOST || '';
const topoApiPort = process.env.TOPO_API_PORT || '443';
const topoBasePath = process.env.TOPO_BASE_PATH;
const topoApiVersion = process.env.TOPO_API_VERSION || '/api/v0.0';
export const SOURCES_API_BASE = `https://${topoApiHost}:${topoApiPort}${topoBasePath}${topoApiVersion.replace(/\/+$/, '')}`;

const apiHost = process.env.API_HOST || 'localhost';
const apiPort = process.env.API_PORT || '5000';
const basePath = process.env.BASE_PATH || '';
const apiVersion = process.env.API_VERSION || '/api/v0.0';
export const SERVICE_PORTAL_API_BASE = `https://${apiHost}:${apiPort}${basePath}${apiVersion.replace(/\/+$/, '')}`;
