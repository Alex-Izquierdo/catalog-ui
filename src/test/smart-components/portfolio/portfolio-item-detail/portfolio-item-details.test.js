import React from 'react';
import thunk from 'redux-thunk';
import { shallow, mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store' ;
import { shallowToJson } from 'enzyme-to-json';
import { MemoryRouter, Route } from 'react-router-dom';
import promiseMiddleware from 'redux-promise-middleware';
import { notificationsMiddleware } from '@red-hat-insights/insights-frontend-components/components/Notifications';

import { ProductLoaderPlaceholder } from '../../../../PresentationalComponents/Shared/LoaderPlaceholders';
import PortfolioItemDetail from '../../../../SmartComponents/Portfolio/portfolio-item-detail/portfolio-item-detail';
import { APPROVAL_API_BASE, TOPOLOGICAL_INVENTORY_API_BASE, CATALOG_API_BASE } from '../../../../Utilities/Constants';
import PortfolioItemDetailToolbar from '../../../../SmartComponents/Portfolio/portfolio-item-detail/portfolio-item-detail-toolbar';
import ItemDetailInfoBar from '../../../../SmartComponents/Portfolio/portfolio-item-detail/item-detail-info-bar';
import ItemDetailDescription from '../../../../SmartComponents/Portfolio/portfolio-item-detail/item-detail-description';

describe('<PortfolioItemDetail />', () => {
  let initialProps;
  let initialState;
  const middlewares = [ thunk, promiseMiddleware(), notificationsMiddleware() ];
  let mockStore;

  const ComponentWrapper = ({ store, children, initialEntries }) => (
    <Provider store={ store }>
      <MemoryRouter initialEntries={ initialEntries }>
        { children }
      </MemoryRouter>
    </Provider>
  );

  beforeEach(() => {
    initialProps = {};
    initialState = {
      approvalReducer: {
        workflows: [{
          label: 'foo',
          value: 'bar'
        }],
        isFetching: false
      },
      portfolioReducer: {
        portfolioItem: {
          id: '123',
          service_offering_source_ref: '123',
          created_at: '123',
          name: 'bar'
        }
      },
      platformReducer: {
        platforms: [{
          id: '123',
          name: 'source'
        }]
      }
    };
    mockStore = configureStore(middlewares);
  });

  afterEach(() => {
    fetchMock.reset();
  });

  it('should render correctly', () => {
    const wrapper = shallow(<PortfolioItemDetail { ...initialProps } />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it.skip('should mount show loader', done => {
    const store = mockStore(initialState);
    fetchMock.getOnce(`${APPROVAL_API_BASE}/workflows`, { data: []});
    fetchMock.getOnce(`begin:${CATALOG_API_BASE}/portfolio_items`, { name: 'foo', id: 'bar' });
    apiClientMock.get(`${TOPOLOGICAL_INVENTORY_API_BASE}/sources`, mockOnce({ body: { data: []}}));
    const wrapper = mount(
      <ComponentWrapper store={ store }>
        <PortfolioItemDetail { ...initialProps } />
      </ComponentWrapper>
    );
    setImmediate(() => {
      expect(wrapper.find(ProductLoaderPlaceholder)).toHaveLength(1);
      done();
    });
  });

  it('should mount load data and render correct components', done => {
    let loadedState = {
      ...initialState,
      portfolioReducer: {
        ...initialState.portfolioReducer,
        selectedPortfolio: {
          name: 'foo',
          id: '321'
        }
      }
    };
    const store = mockStore(loadedState);
    fetchMock.getOnce(`${APPROVAL_API_BASE}/workflows`, { data: []});
    fetchMock.getOnce(`${CATALOG_API_BASE}/portfolio_items/123`, { name: 'foo', id: 'bar' });
    apiClientMock.get(`${TOPOLOGICAL_INVENTORY_API_BASE}/sources`, mockOnce({ body: { data: []}}));

    const wrapper = mount(
      <ComponentWrapper store={ store } initialEntries={ [ '/foo/123' ] }>
        <Route path="/foo/:portfolioItemId" render={ (...args) => <PortfolioItemDetail { ...initialProps } { ...args } /> } />
      </ComponentWrapper>
    );
    setImmediate(() => {
      expect(wrapper.find(ItemDetailInfoBar)).toHaveLength(1);
      expect(wrapper.find(ItemDetailDescription)).toHaveLength(1);
      expect(wrapper.find(PortfolioItemDetailToolbar)).toHaveLength(1);
      done();
    });
  });
});
