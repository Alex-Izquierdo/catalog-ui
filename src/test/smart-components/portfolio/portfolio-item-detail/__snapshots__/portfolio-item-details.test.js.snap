// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PortfolioItemDetail /> should render correctly 1`] = `
<Route>
  <Component />
</Route>
`;
